packages/preview-middleware/src/base/config.ts

```diff
@@ -204,22 +205,34 @@ function getFlexSettings(): TemplateConfig['ui5']['flex'] {
  * @param manifest manifest of the additional target app
  * @param app configuration for the preview
  * @param logger logger instance
+ * @param descriptor descriptor of the additional target app
  */
-export async function addApp(templateConfig: TemplateConfig, manifest: Partial<Manifest>, app: App, logger: Logger) {
+export async function addApp(
+    templateConfig: TemplateConfig,
+    manifest: Partial<Manifest>,
+    app: App,
+    logger: Logger,
+    descriptor?: MergedAppDescriptor
+) {
     const id = manifest['sap.app']?.id ?? '';
+
     app.intent ??= {
         object: id.replace(/\./g, ''),
         action: 'preview'
     };
+
+    const appName = `${app.intent?.object}-${app.intent?.action}`;
     templateConfig.ui5.resources[id] = app.target;
-    templateConfig.apps[`${app.intent?.object}-${app.intent?.action}`] = {
+    templateConfig.apps[appName] = {
         title: (await getI18nTextFromProperty(app.local, manifest['sap.app']?.title, logger)) ?? id,
         description: (await getI18nTextFromProperty(app.local, manifest['sap.app']?.description, logger)) ?? '',
         additionalInformation: `SAPUI5.Component=${app.componentId ?? id}`,
         applicationType: 'URL',
-        url: app.target,
-        applicationDependencies: { manifest: true }
```

Comment:
Did you test this with a Fiori application project? Isnt `applicationDependencies: { manifest: true }`required for apps?

Reply:
I did some tests with Fiori application projects and I found no issues with the changes. I also compared the resulting `flpSandbox.html` without my changes and I also don't see the `applicationDependencies: { manifest: true }`.  But if you think the risk of a regression is too big, I can add it back.

![image](https://github.com/user-attachments/assets/5a300c89-d9d2-4e46-aa52-70d0af0fff0e)

Reply:
Thanks for verifying my concerns. All good with me.